src/components/refactor/rhf-inputs/table-inputs/table-numerical-input.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { TextField, Tooltip } from '@mui/material';
+import { useController, useFormContext } from 'react-hook-form';
+import { useIntl } from 'react-intl';
+
+export const TableNumericalInput = ({
+    name,
+    min,
+    max,
+    style,
+    inputProps,
+    ...props
+}) => {
+    const { trigger } = useFormContext();
+    const {
+        field: { onChange, value, ref },
+        fieldState: { error },
+    } = useController({ name });
+
+    const intl = useIntl();
+
+    const inputTransform = (value) => {
+        if (['-', '.'].includes(value)) return value;
+        return value === null || isNaN(value) ? '' : value.toString();
```

Comment:
you can remplace `value === null` by `value == null` to check undefined too

Reply:
value will not be undefined since it's not supported by react hook form, and I'm not a fan of coercions in comparisons